src/lib/handlePageRequest.ts

```diff
@@ -13,19 +13,18 @@ import {
   withCookieRedirect,
 } from "./middleware";
 
-type TransformFunction = (formData: CacheEntry) => CacheEntry;
+type TransformCallback = (formData: CacheEntry) => CacheEntry;
 
-export type GetFieldManager = (formData: CacheEntry) => FormManager;
+export type DefineFormRulesCallback = (formData: CacheEntry) => FormManager;
```

Comment:
For me, a better name for this would be something like `GetFormManagerFactory` or `FormManagerCacheFactory`. As the function itself returns an instance of `FormManager`, given a cache entry it feels a little weird to call: `const formManager = defineFormRulesCallback(getCache(cookies));` as you're returning an instance of `FormManager` from the function, with subsequent calls to that returned value being on the class `FormManager`. Currently, the function name implies it is returning a form rules object. Essentially, it's a factory function that returns an instance of `FormManager` based on cache entries!

Reply:
I think you're right to raise this!  I guess the tension is that we want to find a name that is readable / makes sense in two places: (1) the form page and (2) the middleware.

On the page, `defineFormRules` is more meaningful than `getFormManagerFactory` because it signifies what you're doing at the start of each page when you instantiate the `FormManager` and `FieldManager` classes (it's a higher level of abstraction).  Equally we're not defining `formRules` in the middleware so it's a sub-optimal name there!

Let's split the two.  How about...

On the page:

```typescript
const **defineFormRules** = ({
  manufacturer,
  model,
  hexId,
}: CacheEntry): FormManager => {
  return new FormManager({ ...
```

In the middleware:

```typescript
export const handlePageRequest = (
  destinationIfValid: string,
  **formManagerFactory: FormManagerFactory,**
  transformCallback: TransformCallback = (formData) => formData
): GetServerSideProps => ...
```

?

Reply:
Yeah good point. Agree with the name in the `handlePageRequest` as it's calling the function to instantiate an instance of `FormManager` and calling methods on it.

Happy with `defineFormRules`.  Only thing I would say is, for me, it's slightly too specific.  I would have probably gone with something like `definePageForm` or something like that as it's not just the rules you're defining in there but also the structure/fields

Reply:
YES!  I like `definePageForm`.  Will make that change and commit 👍🏻

Reply:
Super, please merge once you've made those changes @zackads